lib/apartment/adapters/abstract_adapter.rb

```diff
@@ -224,6 +231,34 @@ def rescuable_exceptions
       def rescue_from
         []
       end
+
+      def extended_cloned_config(tenant)
+        @config.clone.with_indifferent_access.tap do |config|
+          config.merge!(Apartment.db_config_for(tenant))
```

Comment:
i'm not sure we ever want to be merging our specific tenant db config with the current configuration, seems like it would never be useful but be an area for potential problems if someone accidentally doesn't specify all the connection information for a tenant.

can't we just check `use_tenant_specific_db_conf?` and then use either `@config` or `Apartment.db_config_for(tenant)` ?